src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs

```diff
@@ -451,15 +439,7 @@ private INamedTypeSymbol GetPropertyType(string ownerType, string propertyName)
 						{
 							var baseType = type.BaseType;
 
-							if (baseType == null)
-							{
-								var baseTypeName = type?.BaseType?.ToDisplayString();
-								baseType = baseTypeName != null ? FindType(baseTypeName) : null;
-							}
-
-							type = baseType;
-
-							if (type == null || type.SpecialType == SpecialType.System_Object)
+							if (baseType == null || baseType.SpecialType == SpecialType.System_Object)
 							{
 								return null;
 							}
```

Comment:
If `baseType` is null, then `baseTypeName` will also be null, hence `FindType` is never called and `baseType` will remain null.
In which case we'll end up just returning null because `type == null` is true.

So I just simplified things to return null directly if `baseType == null`

But correct me if I'm wrong.